apps/funding-platform/src/access-token/access-token.ts

```diff
@@ -0,0 +1,39 @@
+import { createHmac, randomInt } from "crypto";
+
+export class AccessToken {
+  private hash?: string;
+  private createdAt: Date;
+  constructor(
+    private expiredAt: Date,
+    private amount: number,
+    private secretKey: string
+  ) {
+    this.createdAt = new Date(Date.now());
+  }
+
+  public getCreatedAt() {
+    return this.createdAt;
+  }
+
+  public getExpiredAt() {
+    return this.expiredAt;
+  }
+
+  public getHash() {
+    return this.hash;
+  }
+
+  public generateHash(): string {
```

Comment:
Is there a case where we want to create an `new AccessToken(..)` without calling `.generateHash()`?
I would suspect that whenever we call `new AccessToken(..)` we also want the hash to be generated, in that case I would already generate the hash so we can never have `.getHash() === undefined`